activerecord/lib/active_record/query_cache.rb

```diff
@@ -24,17 +24,10 @@ def uncached(&block)
     end
 
     def self.run
-      connection    = ActiveRecord::Base.connection
-      enabled       = connection.query_cache_enabled
-      connection.enable_query_cache!
-
-      [connection, enabled]
+      ActiveRecord::Base.connection.enable_query_cache!
     end
 
-    def self.complete((connection, enabled))
-      connection.clear_query_cache
-      connection.disable_query_cache! unless enabled
-
+    def self.complete(_)
       unless ActiveRecord::Base.connected? && ActiveRecord::Base.connection.transaction_open?
```

Comment:
Right now the query cache is always disabled at the end of every request. If we instead rely on a callback to disable it when the connection returns to the pool, we'll leak connections with the query cache enabled if a transaction is left open at this point.

I think it's actually fine to say "if your app leaks connections with open transactions, the consequences will be so severe that you won't even notice the query cache problem" and not do anything about this, but I thought it was worth at least pointing out.